gzread.c

```diff
@@ -285,7 +285,7 @@ static size_t gz_read(gz_state *state, void *buf, size_t len) {
     got = 0;
     do {
         /* set n to the maximum amount of len that fits in an unsigned int */
-        n = -1;
+        n = (unsigned)-1;
```

Comment:
Shouldn't this be ```n = UINT_MAX;``` as assigning negative number to unsigned variable can also cause warning.